### PR TITLE
272 create time series hmtl for water levels at obervation stations

### PR DESCRIPTION
**Pull request by GundulaW**
Waiting for coordination with Maarten to finalize this

### CODE REVIEW

**Comment by GundulaW** [comment]
This requires attached site.toml
[site.zip](https://github.com/Deltares-research/FloodAdapt/files/13676573/site.zip)

File: flood_adapt/api/output.py [hunk 1]
```diff
@@ -43,6 +43,45 @@ def get_roads(name: str, database: IDatabase) -> gpd.GeoDataFrame:
...
+    # Get the direct_impacts objects from the scenario
+    hazard = database.get_scenario(name).direct_impacts.hazard
+
+    # Check if the scenario has run
+    if not hazard.sfincs_has_run_check():
+        raise ValueError(
+            f"Scenario {name} has not been run. Please run the scenario first."
+        )
+
+    database_path = Path(database.input_path).parent
```
Comment (dladrichem) [inline flood_adapt/api/output.py:70]: The database now also stores the static and output paths so this is not needed anymore

File: flood_adapt/api/output.py [hunk 2]
```diff
@@ -43,6 +43,45 @@ def get_roads(name: str, database: IDatabase) -> gpd.GeoDataFrame:
...
+    output_path = database_path.joinpath("output", "Scenarios", hazard.name)
+    gdf = database.get_obs_points()
+    timeseries_output_path = []
```
Comment (dladrichem) [inline flood_adapt/api/output.py:73]: It's a bit cleaner to do this as a list comprehension:
gdf["html"] = [ str(output_path.joinpath("Flooding", f"{station}_timeseries.html")) for station in gdf.name ]

File: flood_adapt/integrator/sfincs_adapter.py [hunk 3]
```diff
@@ -429,12 +429,61 @@ def set_config_spw(self, spw_name: str):
     def turn_off_bnd_press_correction(self):
         self.sf_model.set_config("pavbnd", -9999)
 
+    def add_obs_points(self):
+        """add observation points provided in the site toml to SFINCS model"""
+
+        if self.site.attrs.obs_point is not None:
```
Comment (dladrichem) [inline flood_adapt/integrator/sfincs_adapter.py:435]: Can't you use the get_obs_points in the dbs_controller for both of these (add_obs_points and read_zs_points)?

File: flood_adapt/object_model/hazard/hazard.py [hunk 4]
```diff
@@ -663,6 +672,105 @@ def postprocess_sfincs(self):
...
+                    # check if observation station has a tide gauge ID
+                    # if yes to both download tide gauge data and add to plot
+                    if isinstance(self.site.attrs.obs_point[ii].ID, int):
+                        df_gauge = HistoricalNearshore.download_wl_data(
+                            station_id=self.site.attrs.obs_point[ii].ID,
+                            start_time_str=self.event.attrs.time.start_time,
+                            stop_time_str=self.event.attrs.time.end_time,
+                            units=UnitTypesLength(gui_units),
+                        )
+                        import plotly.graph_objects as go
```
Comment (dladrichem) [inline flood_adapt/object_model/hazard/hazard.py:752]: If it doesn't interfere with other imports (which it shouldn't), just put them on top

File: tests/test_database/charleston/static/site/site.toml [hunk 5]
```diff
@@ -73,6 +73,8 @@ footprints_dmg_bins = [0.00001, 15000, 50000, 100000, 250000]
 footprints_dmg_colors = ['#FFFFFF', '#FEE9CE', '#FDBB84', '#FC844E', '#E03720', '#860000']
 svi_bins = [0.05, 0.2, 0.4, 0.6, 0.8]
 svi_colors = ['#FFFFFF', '#FEE9CE', '#FDBB84', '#FC844E', '#E03720', '#860000']				   																			   
+benefits_bins = [0, 0.01, 1000000, 10000000, 50000000]
```
Comment (dladrichem) [inline tests/test_database/charleston/static/site/site.toml:76]: Is this supposed to be in here?

File: flood_adapt/api/output.py (hunk 1 above)
Comment (GundulaW) [reply flood_adapt/api/output.py:70]: ok, also added that in IDatabase so the type helper can pick this up

File: flood_adapt/api/output.py (hunk 2 above)
Comment (GundulaW) [reply flood_adapt/api/output.py:73]: Done

File: flood_adapt/integrator/sfincs_adapter.py (hunk 3 above)
Comment (GundulaW) [reply flood_adapt/integrator/sfincs_adapter.py:435]: But how does the SFINCS adapter know about the database?

Comment (GundulaW) [reply flood_adapt/integrator/sfincs_adapter.py:435]: I'll make a separate issue out of it

**Pull request merged**